core/util/http.py

```diff
@@ -252,16 +252,19 @@ def _request_with_timeout(
         if "data" in kwargs and isinstance(kwargs["data"], str):
             kwdata: str = kwargs["data"]
             kwargs["data"] = kwdata.encode("utf8")
-        if "headers" in kwargs:
-            headers = kwargs["headers"]
-            new_headers = {}
-            for k, v in list(headers.items()):
-                if isinstance(k, str):
-                    k = k.encode("utf8")
-                if isinstance(v, str):
-                    v = v.encode("utf8")
-                new_headers[k] = v
-            kwargs["headers"] = new_headers
+
+        headers = kwargs.get("headers", {})
+        # Set a user-agent if not already present
+        if "User-Agent" not in headers:
+            headers["User-Agent"] = "Palace Manager/1.x.x"
```

Comment:
How much work would it be to pull in the version (and potentially revision)?  Not sure if this is out of scope.

Reply:
We currently don't have the concept of numbered versions (like semver), we only use a commit hash. Hence we are using this stopgap example for now.

Reply:
This should perhaps reference the version number that gets set in the configuration so it returns the same info as the heartbeat endpoint. 

https://github.com/ThePalaceProject/circulation/blob/dd910452b4e7883f580df1854a38977a9fc767ef/core/config.py#L522

Reply:
@jonathangreen It was my suggestion to include a generic version string until we could discuss any security implications. But, if you're good with it, I have no objection.

Also, good to see you here!

Reply:
Updated to use app_version.